internal/driver/driver.go

```diff
@@ -633,23 +634,31 @@ func (d *Driver) startStreaming(device *Device) errors.EdgeX {
 		return errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf(
 			"failed to start video streaming for device %s", device.name), err)
 	}
+	startErrs := make(chan errors.EdgeX)
 	d.wg.Add(1)
 	go func() {
 		select {
 		case err := <-errChan:
 			device.StopStreaming(err)
-			d.lc.Errorf("the video streaming process for device %s has stopped, error: %s", device.name, err)
+			d.lc.Debugf("the video streaming process for device %s has stopped", device.name)
+			startErrs <- errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("the video streaming process for device %s has stopped, error: %s", device.name, err), err)
 			d.wg.Done()
 			return
 		case <-device.ctx.Done():
 			if err := device.transcoder.Stop(); err != nil {
 				d.lc.Errorf("failed to stop video streaming for device %s, error: %s", device.name, err)
+				startErrs <- errors.NewCommonEdgeX(errors.KindServerError, fmt.Sprintf("failed to stop video streaming for device %s, error: %s", device.name, err), err)
+				d.wg.Done()
+				return
 			}
 			d.lc.Debugf("the video streaming process for device %s has stopped", device.name)
 			d.wg.Done()
 			return
 		}
 	}()
+	if startErrs != nil {
+		return <-startErrs
+	}
```

Comment:
This will block forever if no errors. Needs a timeout, so something like this. Need to determine appropriate timeout value  
```suggestion
        for {
          select {
          case <-time.After(time.Second)
             break
          case err  <-startErrs
	     return err
        }
```